sympy/core/expr.py

```diff
@@ -3668,12 +3668,9 @@ def round(self, n=None):
                     'Expected a number but got %s:' % func_name(x))
         elif x in (S.NaN, S.Infinity, S.NegativeInfinity, S.ComplexInfinity):
             return x
-        if not x.is_extended_real:
```

Comment:
This is meant to handle things like `pi/2 + e*I`...why are you deleting this?

Reply:
Sorry, I thought I had verified that this wasn't an extended_real and that the tests passed.